### DISCUSSION TITLE
Code questions

### DISCUSSION THREAD

**Discussion by tbertels** [mpc-qt/mpc-qt#218] [General]
Hi,

What does [`PlaybackManager::mpvw_playlistChanged`](https://github.com/mpc-qt/mpc-qt/blob/b5c28e19ef2ea470fd4c43f731e8bc6be44bd5d5/manager.cpp#L962) do? It never seems to get called.
When I check the value of `playlist.count()` in [`MpvObject::ctrl_hookEvent`](https://github.com/mpc-qt/mpc-qt/blob/b5c28e19ef2ea470fd4c43f731e8bc6be44bd5d5/mpvwidget.cpp#L682), it's either 0 or 1, thus the above.

https://mpv.io/manual/master/#command-interface-playlist mentions for the `playlist` property: 

> Playlist, current entry marked. Currently, the raw property value is useless.

But on unload, it seems this property only contains the current item.

Here's an example of dumping the playlist variable in the log in `MpvObject::ctrl_hookEvent`:

`[5.413] [mpvwidget] playlist: [{"current":true, "filename":/home/user/Downloads/sample-video.mkv, "id":2}]`

Notice the `"id":2`, it was the third video of the playlist, with more videos after it.

Relevant commits:
80bfe6795a71faee1ea23f5a5819810d57331340
ab0aa5133b30a8e30479e1e24c7dcccc3b39454b

**Comment by cmdrkotori** [comment]
Archives have more than one file inside them.  For example, comic book files (cbz,cbr) are simply archives with images inside them.  I've also heard that some people store entire audio cds in zip files.

When an archive is opened, mpv's internal playlist is populated with the items in the archive.  So the code in mpc-qt is supposed to take that and insert it over the top of the zip file entry in the active playlist.  Unfortunately, one can't easily tell if the archive won't be full of unplayable garbage until after it's opened.
